src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -1657,10 +1663,21 @@ static int do_map(int argc, const char *argv[], Config *cfg, bool reconnect)
     }
   }
 
-  if (!cfg->snapname.empty()) {
+  if (!cfg->snapid.empty()) {
+    uint64_t snap_id = boost::lexical_cast<uint64_t>(cfg->snapid);
```

Comment:
Since you have chosen to go with `snapid`, stick to it:
```suggestion
    uint64_t snapid = boost::lexical_cast<uint64_t>(cfg->snapid);
```